book/configuration_preview.md

```diff
@@ -0,0 +1,669 @@
+   technique.
+
+## Common Configuration Tasks in `config.nu`
+
+The primary purpose of `config.nu` is to:
+
+- Override default Nushell settings in the `$env.config` record
+- Define or import custom commands
+- Run other startup tasks
+
+::: tip
+Some users will prefer a "monolithic" configuration file with most or all startup tasks in one place. `config.nu` can be used for this purpose.
+
+Other users may prefer a "modular" configuration where each file handles a smaller, more focused set of tasks. Files in the autoload dirs can be used to create this experience.
+:::
+
+### Changing Settings in the `$env.config` Record
+
+The primary mechanism for changing Nushell's behavior is the `$env.config` record. While this record is accessed as an environment variable, unlike most other variables it is:
+
+- Not inherited from the parent process. Instead, it is populated by Nushell itself with certain defaults.
+- Not exported to child processes started by Nushell.
+
+To examine the current settings in `$env.config`, just type the variable name:
+
+```nu
+$env.config
+```
+
+::: tip
+Since Nushell provides so many customization options, it may be better to send this to a pager like:
+
+```nu
+$env.config | table -e | less -R
+# or, if bat is installed:
+$env.config | table -e | bat -p
+```
+
+:::
+
+An appendix documenting each setting will be available soon. In the meantime, abbreviated documentation on each setting can be
+viewed in Nushell using:
+
+```nu
+config nu --sample | nu-highlight | bat
+# or
+config nu --sample | nu-highlight | less -R
+```
+
+To avoid overwriting existing settings, it's best to simply assign updated values to the desired configuration keys, rather than the entire `config` record. In other words:
+
+::: warning Wrong
+
+```nu
+$env.config = {
+  show_banner: false
+}
+```
+
+This would reset any _other_ settings that had been changed, since the entire record would be overwritten.
+:::
+
+::: tip Right
+
+```nu
+$env.config.show_banner = false
+```
+
+This changes _only_ the `show_banner` key/value pair, leaving all other keys with their existing values.
+:::
+
+Certain keys are themselves also records. It's okay to overwrite these records, but it's best-practice
+to set all values when doing so. For example:
+
+```nu
+$env.config.history = {
+  file_format: sqlite
+  max_size: 1_000_000
+  sync_on_enter: true
+  isolation: true
+}
+```
+
+### Remove Welcome Message
+
+:::note
+This section is linked directly from the banner message, so it repeats some information from above.
+:::
+
+To remove the welcome message that displays each time Nushell starts:
+
+1. Type `config nu` to edit your configuration file.
+2. If you receive an error regarding the editor not being defined:
+
+   ```nu
+   $env.config.buffer_editor = <path to your preferred editor>
+   # Such as:
+   $env.config.buffer_editor = "code"
+   $env.config.buffer_editor = "vi"
```

Comment:
It would be good to add an example here of an editor with parameters like this one that recently came up on discord.
```nushell
$env.config.buffer_editor: ["emacsclient", "-s", "light", "-t"],
```

Reply:
Good point - I didn't even know about that one!